xblock/fields.py

```diff
@@ -1013,22 +1042,22 @@ def scope_key(instance, xblock):
     if instance.scope.user == UserScope.NONE or instance.scope.user == UserScope.ALL:
         pass
     elif instance.scope.user == UserScope.ONE:
-        scope_key_dict['user'] = unicode(xblock.scope_ids.user_id)
+        scope_key_dict['user'] = str(xblock.scope_ids.user_id)
     else:
         raise NotImplementedError()
 
     if instance.scope.block == BlockScope.TYPE:
-        scope_key_dict['block'] = unicode(xblock.scope_ids.block_type)
+        scope_key_dict['block'] = str(xblock.scope_ids.block_type)
```

Comment:
I understand the goal. Another goal is, don't drive the developers crazy as they try to work in this polyglot world. It might be better to have an obvious indicator that things are 2/3-complicated.  If I dive into a repo and see "str", how do I determine what it means? I have to keep my eye on the top of the file to know what paradigm I'm using: pure 2, pure 3, 2/3 with futurize, "2/3 with futurize but forgot to import str", etc.  I think six has an advantage that its polyglot nature is clearer with a smaller search radius.